scan-command/src/main/java/io/ballerina/scan/internal/StaticCodeAnalyzer.java

```diff
@@ -18,24 +18,43 @@
 
 package io.ballerina.scan.internal;
 
+import io.ballerina.compiler.syntax.tree.CheckExpressionNode;
 import io.ballerina.compiler.syntax.tree.ModulePartNode;
 import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
 import io.ballerina.compiler.syntax.tree.SyntaxTree;
 import io.ballerina.projects.Document;
+import io.ballerina.scan.ScannerContext;
 
 /**
  * {@code StaticCodeAnalyzer} contains the logic to perform core static code analysis on Ballerina documents.
  *
  * @since 0.1.0
  * */
 class StaticCodeAnalyzer extends NodeVisitor {
+    private final Document document;
     private final SyntaxTree syntaxTree;
+    private final ScannerContext scannerContext;
 
     StaticCodeAnalyzer(Document document, ScannerContextImpl scannerContext) {
+        this.document = document;
```

Comment:
I'm npt sure if I raised this previously, but do we have to create a new StaticCodeAnalyzer for each document? Can't we do by project instead? And within this we analyze all the documents of the project?

Reply:
Since we are visiting the syntax tree of each document, we will have to create a new node visitor for each document with their syntax tree passed to the node visitors visit method. 

Since a new node visitor is to be created, I've extended it through the StaticCodeAnalyzer class,  so a new StaticCodeAnalyzer is created for each document.